tsl/src/bgw_policy/job.c

```diff
@@ -159,21 +184,66 @@ execute_drop_chunks_policy(int32 job_id)
 
 	/* Get the arguments from the drop_chunks_policy table */
 	args = ts_bgw_policy_drop_chunks_find_by_job(job_id);
-
+	table_relid = ts_hypertable_id_to_relid(args->fd.hypertable_id);
 	if (args == NULL)
 		ereport(ERROR,
 				(errcode(ERRCODE_TS_INTERNAL_ERROR),
 				 errmsg("could not run drop_chunks policy #%d because no args in policy table",
 						job_id)));
 
-	ts_chunk_do_drop_chunks(ts_hypertable_id_to_relid(args->fd.hypertable_id),
-							IntervalPGetDatum(&args->fd.older_than),
-							0,
-							INTERVALOID,
-							InvalidOid,
-							args->fd.cascade,
-							args->fd.cascade_to_materializations,
-							LOG);
+	if (args->fd.older_than.is_time_interval)
+	{
+		ts_chunk_do_drop_chunks(table_relid,
+								IntervalPGetDatum(&args->fd.older_than.time_interval),
+								(Datum) 0,
+								INTERVALOID,
+								InvalidOid,
+								args->fd.cascade,
+								args->fd.cascade_to_materializations,
+								LOG);
+	}
+	else
+	{
+		hcache = ts_hypertable_cache_pin();
```

Comment:
can we also create an api in interval.c/.h takes a ts_interval and a hypertable_id and returns a datum with internal type or something. Not sure here.

Reply:
it won't mbe much simpler, but can factor out the lookup call into a dedicated interval function?

Reply:
DISCUSS

Reply:
Some of the complexity comes from the fact that several calls are required to get the open dimension partitioning type of the hypertable. Perhaps we could add another function similar to ts_dimension_get_partition_type():
 `ts_hypertable_get_open_partition_type(Oid relid)`